pkgs/by-name/xe/xephem/package.nix

```
@@ -0,0 +1,94 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  makeDesktopItem,
+  copyDesktopItems,
+  installShellFiles,
+  motif,
+  openssl,
+  groff,
+  xorg,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xephen";
+  version = "4.2.0";
+
+  src = fetchFromGitHub {
+    owner = "XEphem";
+    repo = "XEphem";
+    rev = version;
+    hash = "sha256-TuzXrWoJOAHg31DrJObPcHBXgtqR/KWKFRsqddPzL4c=";
+  };
+
+  nativeBuildInputs = [
+    motif
+    openssl
+    groff
+    xorg.libXmu
+    xorg.libXext
+    xorg.libXt
+    copyDesktopItems
+    installShellFiles
+  ];
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/XEphem/XEphem/commit/30e14f685ede015fcd8985cd83ee6510f93f0073.patch";
+      hash = "sha256-wNoLjR6xEl56ZA6FLBS2xtySeDEYXTCA8j4Z5JIrF6k=";
+    })
+  ];
+
+  postPatch = ''
+    cd GUI/xephem
+    substituteInPlace xephem.c splash.c --replace-fail '/etc/XEphem' '${placeholder "out"}/etc/XEphem'
+  '';
+
+  doCheck = true;
+
+  checkFlags = "-C ../../tests";
+
+  checkTarget = "run-test";
+
+  installPhase = ''
+    runHook preInstall
+    installBin xephem
+    mkdir -p $out/share/xephem
+    cp -R auxil $out/share/xephem/
+    cp -R catalogs $out/share/xephem/
+    cp -R fifos $out/share/xephem/
+    cp -R fits $out/share/xephem/
+    cp -R gallery $out/share/xephem/
+    cp -R help $out/share/xephem/
+    cp -R lo $out/share/xephem/
+    mkdir $out/etc
+    echo "XEphem.ShareDir: $out/share/xephem" > $out/etc/XEphem
+    installManPage xephem.1
+    install -Dm644 XEphem.png -t $out/share/icons
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "xephem";
+      exec = "xephem";
+      icon = "XEphem";
+      desktopName = "XEphem";
+      categories = [
+        "Science"
+        "Astronomy"
+      ];
+    })
+  ];
+
+  meta = {
+    description = "The Serious Interactive Astronomical Software Ephemeris";
+    mainProgram = "xephem";
+    homepage = "https://xephem.github.io/XEphem/Site/xephem.html";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ EstebanMacanek ];
+    platforms = lib.platforms.unix;
```

```suggestion
    platforms = lib.platforms.unix;
    # Makefiles hardcode unprefixed tools everywhere
    broken = stdenv.buildPlatform != stdenv.hostPlatform;
```
Currently the package does not cross-compile, so we should mark it as such.

---

No, you would need to replace all occurences of `cc` / `ar` / `ranlib` manually, since some Makefiles do not use the `CC` variables etc.

https://github.com/XEphem/XEphem/blob/main/libz/Makefile